src/tictactoe/ai.clj

```diff
@@ -0,0 +1,49 @@
+(ns tictactoe.ai
+  (:require [tictactoe.board :refer [place-move winner game-over? available-spaces]]))
+
+(defn ^:private take-turn
+  [depth computer opponent]
+  (if (not (zero? (mod depth 2)))
+    computer
+    opponent))
+
+(defn ^:private scores
+  [board computer depth]
+  (let [winning-player (winner board)]
+    (cond
+      (= winning-player computer) (- 10 depth)
+      (nil? winning-player) 0
+      :else (- depth 10))))
+
+(declare moves-and-scores)
+
+(defn ^:private min-or-max 
+  [board-state depth]
+  (if (not (zero? (mod depth 2)))
+    (apply min (vals board-state))
+    (apply max  (vals board-state))))
+
+(defn ^:private get-scores
+  [board computer opponent depth]
+  (let [board-state board]
+    (if (game-over? board-state)
+      (scores board-state computer depth)
+      (min-or-max (moves-and-scores board-state computer opponent (inc depth)) depth))))
+
+(def memoize-get-scores (future (memoize get-scores)))
```

Comment:
I don't think this needs to be a `future`. `future` just takes some body of work and executes it on another thread. The "work" here is just calling `memoize` on `get-scores`. If I recall correctly, `memoize` just takes a function, wraps it in some state, and returns a new function which is memoized. It almost certainly takes no time at all to do that.